stdlib/public/Darwin/Foundation/String.swift

```diff
@@ -49,15 +102,82 @@ extension String : _ObjectiveCBridgeable {
     self._forceBridgeFromObjectiveC(x, result: &result)
     return result != nil
   }
+  
+  @_effects(readonly)
+  @inline(__always)
+  private static func _bridgeToSmall(
+    _ source: NSString,
+    _ len: Int,
+    _ encoding: CFStringBuiltInEncodings) -> String? {
+    return String(unsafeUninitializedCapacity: 15) { (ptr, outCount) in
+      let converted = _getBytes(
+        source,
+        CFRange(location: 0, length: len),
+        encoding.rawValue,
+        UnsafeMutableRawPointer(ptr.baseAddress!).assumingMemoryBound(to:
+          UInt8.self),
+        15, //maxBufLen
+        &outCount
+      )
+      
+      return converted == len
+    }
+  }
+  
+  @_effects(readonly)
+  @inline(__always)
+  private static func _bridgeTagged(_ source: NSString) -> String? {
+      if _isObjCTaggedPointer(source) {
+        return _bridgeToSmall(source, _length(source), CFStringBuiltInEncodings.ASCII)
+      }
+      return nil
+  }
+  
+  @_effects(readonly)
+  private static func _unconditionallyBridgeFromObjectiveC_nonTagged(
+    _ source: NSString
+    ) -> String {
+    let len = _length(source)
+    let sourceClass:AnyClass = _getClass(source)
+    
+    //Only try regular CF-based NSStrings for now due to things like
+    //NSLocalizedString assoc objects and NSAttributedString content proxies
+    //Also only try Strings we believe will fit into a SmallString for now
+    //In the future we should do mutable __NSCFStrings of any length here.
+    if sourceClass == nscfClass,
+      len <= 15,
+      let eager = _bridgeToSmall(source, len, CFStringBuiltInEncodings.UTF8) {
+      return eager
+    }
+    
+    let immutableCopy = (sourceClass == nscfConstantClass) ?
+      source :
+      _copyString(source)
+    
+    //mutable->immutable might make it start being a tagged pointer
+    if let result = _bridgeTagged(immutableCopy) {
+      return result
+    }
+    
+    return _bridgeCocoaStringLazily(immutableCopy, sourceClass, len)
+  }
 
   @_effects(readonly)
   public static func _unconditionallyBridgeFromObjectiveC(
     _ source: NSString?
   ) -> String {
-    // `nil` has historically been used as a stand-in for an empty
-    // string; map it to an empty string.
-    if _slowPath(source == nil) { return String() }
-    return String(source!)
+    guard let source = source else {
+      // `nil` has historically been used as a stand-in for an empty
+      // string; map it to an empty string.
+      return String()
+    }
+    
+    
+    if let result = _bridgeTagged(source) {
```

Comment:
The neat thing about this approach is that it completely eliminates special handling of eager bridging from the Swift side of the bridge. It's just a thing anyone can do with the new String initializer, which we agreed is fine to propose as API.